accepted/0000-hub-online-synchronization.md

```diff
@@ -0,0 +1,402 @@
+
+The new Hub synchronization would be focused on HUB online deployments to allow scale SUSE Manager infrastructure. The only focus of this solution would be to work in scale and make it easier to use.
+
+
+# Detailed design
+[design]: #detailed-design
+
+The solution focuses on re-using the repo-sync mechanism to synchronize channels from Hub server to peripheral servers. It will be described in several steps that aim to address how it will be used and how we can technically support it.
+
+The main focus is to develop the integration between HUB server and peripherals through API only.
+For the Hub/peripheral communication an new API might be useful to have a clear separation. This is needed when we want to make this API organization independent.
+In the existing API the calling user is only allowed to operate on his own organization any maybe utilize "Vendor Channel" when he is a "SUSE Manager Administrator".
+
+Configuration should be pushed from the HUB to peripherals, to avoid the need of direct management of the peripherals.
+
+For now this solution is focused on software channels only, but can be extended to synchronize more data types.
+
+
+## Define connection between HUB and Peripherals
+
+Each peripheral server can only have 1 Hub server (main). This will avoid dealing with problems like channel label conflicts between multiple main servers.
+Configure a hub server on a peripheral server will block access to a set of menu items like: "Admin" -> "Setup wizard" -> "Organization Credentials" and "Products" and "PAYG Connection" (similar to what we already have for ISSv1).
+We cannot add Hub connection if SCC credentials are defined, they should be mutually excluded.
+
+### Alternative 1:
+We can follow a similar approach to what exists on ISSv1. On the hub side we can define multiple peripheral servers to connect to by providing the FQDN and an authentication token.
+On the peripheral side we also need to define the Hub server FQDN and it would generate an Authentication token for he HUB.
+
+To define this connection the existing API namespace should be re-use (`sync.master` and `sync.slave`), with some adaptations to cope with the new requirements. Example of this is the need to define an authentication method between servers to the connection more secure.
+
+During development phase we should consider if we can join both namespaces in one name `hub`, for example.
+
+
+### Alternative 2:
+
+We need to establish a secure connection between Hub and Peripheral Server.
+For this we create a Peripheral configuration on the Hub server.
+It mainly defines the FQDN of the Peripheral and generate credentials to authenticate from the Hub.
+The Root CA must be configured as well when the peripheral server is already setup.
+
+This configuration can be exported and needs to be transferred to the Peripheral Server.
+This configuration include:
+   - the Root CA of the Hub
```

Comment:
We are providing the Root CA from hub to be setup on the peripheral.  Do you think that HUB would need to also get the peripheral CA, in case it's different? I am thinking about the SSL connection to the API, for example.

Reply:
Any outcome here would probably also help for Kubernetes case where it may be hard to share the same RootCA across clusters if using `cert-manager`. We'll face similar issues for proxies on Kubernetes too.